### ISSUE TITLE
Linux kernel 5.11.15 compatibility issue with RFKILL_EVENT_SIZE_V1

### ISSUE THREAD

**Issue by atoussaint76** [systemd/systemd#19349]
systemd version 248
 > …

**Used distribution**
Linux from Scratch with the following modification:
Linux kernel headers 5.11.15 in replacement of kernel headers 5.11.10
gcc-10.3.0 in replacement of gcc-10.2.0
systemd-248 in replacement of systemd 247
All the softwares built in a chroot jail.

**Linux kernel version used** (`uname -a`)

`Linux ryzenbox 5.11.11 #2 SMP Sun Apr 4 05:07:03 EDT 2021 x86_64 GNU/Linux`

**CPU architecture issue was seen on**
x86_64

**Expected behaviour you didn't see**
Successful compilation.

**Unexpected behaviour you saw**
A warning on a signed comparison between ssize_t (long int) and long unsigned int regarding the size of RFKILL_EVENT_SIZE_V1. I investigated the issue in the rfkill.h Linux kernel header and RFKILL_EVENT_SIZE_V1 is set to the size of the rfkill_event struct:
`#define RFKILL_EVENT_SIZE_V1    sizeof(struct rfkill_event)` on line 172 of linux/rfkill.h.

The struct itself is between line 93 and 99.

**Steps to reproduce the problem**
install -vd buildlfs; cd buildlfs
svn co svn://svn.linuxfromscratch.org/ALFS/jhalfs/trunk jhalfs
svn co svn://svn.linuxfromscratch.org/LFS/trunk/BOOK/ lfs
svn co svn://svn.linuxfromscratch.org/BLFS/trunk/BOOK/ blfs
wget https://ftp.gnu.org/gnu/gcc/gcc-10.3.0/gcc-10.3.0.tar.xz
wget https://cdn.kernel.org/pub/linux/kernel/v5.x/linux-5.11.15.tar.xz
wget https://github.com/systemd/systemd/archive/v248/systemd-248.tar.gz
md5sum gcc-10.3.0.tar.xz
md5sum linux-5.11.15.tar.xz
md5sum systemd-248.tar.gz
cd lfs
edit package.ent (vim or emacs or any sufficient text editor), update versions of the afformentioned package and their md5sum
edit patches.ent, remove any patch for systemd upstream.
edit chapter03/patches.xml remove any patch for systemd upstream.
edit chapter08/systemd.xml remove the patch line as well as the sed command.
cd ../jhalfs
make (display an ncurses menu with various settings)
3 important settings: distribution --> LFS systemd
Release --> Working copy
Path --> /home/`<username>`/buildlfs/lfs
any other settings, adjust as you see fit.

**Additional program output to the terminal or log subsystem illustrating the issue**

```
FAILED: systemd-rfkill.p/src_rfkill_rfkill.c.o 
cc -Isystemd-rfkill.p -I. -I.. -Isrc/basic -I../src/basic -Isrc/fundamental -I../src/fundamental -Isrc/systemd -I../src/systemd -I../src/libsystemd/sd-bus -I../src/libsystemd/sd-device -I../src/libsystemd/sd-event -I../src/libsystemd/sd-hwdb -I../src/libsystemd/sd-id128 -I../src/libsystemd/sd-journal -I../src/libsystemd/sd-netlink -I../src/libsystemd/sd-network -I../src/libsystemd/sd-resolve -Isrc/shared -I../src/shared -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=gnu99 -O3 -Wno-unused-parameter -Wno-missing-field-initializers -Wno-unused-result -Wno-format-signedness -Werror=undef -Wlogical-op -Wmissing-include-dirs -Wold-style-definition -Wpointer-arith -Winit-self -Wfloat-equal -Wsuggest-attribute=noreturn -Werror=missing-prototypes -Werror=implicit-function-declaration -Werror=missing-declarations -Werror=return-type -Werror=incompatible-pointer-types -Werror=format=2 -Wstrict-prototypes -Wredundant-decls -Wmissing-noreturn -Wimplicit-fallthrough=5 -Wshadow -Wendif-labels -Wstrict-aliasing=2 -Wwrite-strings -Werror=overflow -Werror=shift-count-overflow -Werror=shift-overflow=2 -Wdate-time -Wnested-externs -Wno-maybe-uninitialized -ffast-math -fno-common -fdiagnostics-show-option -fno-strict-aliasing -fvisibility=hidden -fstack-protector -fstack-protector-strong --param=ssp-buffer-size=4 -ffunction-sections -fdata-sections -Werror=shadow -include config.h -O2 -mno-default -march=x86-64 -mtune=generic -mfpmath=sse -malign-double -m128bit-long-double -malign-data=cacheline -msseregparm -mstackrealign -mpreferred-stack-boundary=4 -mincoming-stack-boundary=4 -mmmx -msse -msse2 -msse3 -mssse3 -msse4 -msse4.1 -msse4.2 -mavx -mvzeroupper -maccumulate-outgoing-args -msse2avx -maes -mpopcnt -mpclmul -free -fira-algorithm=CB -fsched-pressure -ftree-vectorize -fweb --param=l1-cache-line-size=64 --param=l1-cache-size=32 --param=l2-cache-size=3072 -MD -MQ systemd-rfkill.p/src_rfkill_rfkill.c.o -MF systemd-rfkill.p/src_rfkill_rfkill.c.o.d -o systemd-rfkill.p/src_rfkill_rfkill.c.o -c ../src/rfkill/rfkill.c
../src/rfkill/rfkill.c: In function ‘load_state’:
../src/rfkill/rfkill.c:180:15: warning: comparison of integer expressions of different signedness: ‘ssize_t’ {aka ‘long int’} and ‘long unsigned int’ [-Wsign-compare]
  180 |         if (l < RFKILL_EVENT_SIZE_V1)
      |               ^
../src/rfkill/rfkill.c: In function ‘run’:
../src/rfkill/rfkill.c:338:23: warning: comparison of integer expressions of different signedness: ‘ssize_t’ {aka ‘long int’} and ‘long unsigned int’ [-Wsign-compare]
  338 |                 if (l < RFKILL_EVENT_SIZE_V1)
      |                       ^
In file included from ../src/basic/macro.h:491,
                 from ../src/basic/alloc-util.h:9,
                 from ../src/rfkill/rfkill.c:13:
../src/rfkill/rfkill.c:339:70: error: format ‘%d’ expects argument of type ‘int’, but argument 8 has type ‘long unsigned int’ [-Werror=format=]
  339 |                         return log_error_errno(SYNTHETIC_ERRNO(EIO), "Short read of struct rfkill_event: (%zd < %d)",
      |                                                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/basic/log.h:195:86: note: in definition of macro ‘log_full_errno’
  195 |                         ? log_internal(_level, _e, PROJECT_FILE, __LINE__, __func__, __VA_ARGS__) \
      |                                                                                      ^~~~~~~~~~~
../src/rfkill/rfkill.c:339:32: note: in expansion of macro ‘log_error_errno’
  339 |                         return log_error_errno(SYNTHETIC_ERRNO(EIO), "Short read of struct rfkill_event: (%zd < %d)",
      |                                ^~~~~~~~~~~~~~~
../src/rfkill/rfkill.c:339:114: note: format string is defined here
  339 |                         return log_error_errno(SYNTHETIC_ERRNO(EIO), "Short read of struct rfkill_event: (%zd < %d)",
      |                                                                                                                 ~^
      |                                                                                                                  |
      |                                                                                                                  int
      |                                                                                                                 %ld
cc1: some warnings being treated as errors

```
Also, see the comments between lines 132 and 170 of the linux/rfkill.h kernel header.

Thanks,
Alain

**Comment by yuwata** [comment]
Already fixed by ab1aa6368a883bce88e3162fee2bea14aacedf23.

**Comment by atoussaint76** [comment]
Thanks,

I will redo a build when systemd-249 get out.

Alain
